src/store/modules/user/actions.ts

```diff
@@ -222,6 +225,101 @@ const actions: ActionTree<UserState, RootState> = {
       console.error(error);
     }
     return []
+  },
+
+  async setProductIdentificationPref({ commit, state }, payload) {
+    let prefValue = JSON.parse(JSON.stringify(state.productIdentificationPref))
+    const eComStoreId = (state.currentEComStore as any).productStoreId
+
+    const fromDate = UserService.getProductIdentificationPrefFromDate(eComStoreId)
+
+    // when selecting none as ecom store, not updating the pref as it's not possible to save pref with empty productStoreId
+    if(!(state.currentEComStore as any).productStoreId || !fromDate) {
+      showToast(translate('Unable to update product identifier preference'))
+      commit(types.USER_PREF_PRODUCT_IDENT_CHANGED, prefValue)
+      return;
+    }
+
+    prefValue[payload.id] = payload.value
+
+    const params = {
+      "fromDate": fromDate,
+      "productStoreId": eComStoreId,
+      "settingTypeEnumId": "PRODUCT_STORE_PREF",
```

Comment:
We should think of a better name (`PRODUCT_STORE_PREF`), this is already stored in ProductStoreSetting

Reply:
Sir, changed the name to `PRODUCT_IDENTIFIER_PREF`